src/librustc/ty/layout.rs

```diff
@@ -2259,3 +2265,346 @@ impl<'a, 'gcx> HashStable<StableHashingContext<'a>> for LayoutError<'gcx>
         }
     }
 }
+
+pub trait FnTypeExt<'tcx, C> {
+    fn of_instance(cx: &C, instance: &ty::Instance<'tcx>) -> Self
+    where
+        C: LayoutOf<Ty = Ty<'tcx>, TyLayout = TyLayout<'tcx>>
```

Comment:
you could move that bound to the trait declaration instead of having it on the methods, that way you don't have to repeat it